src/main/java/com/google/sps/servlets/CreateUserIdServlet.java

```diff
@@ -47,11 +47,29 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         String userId = userVerifier.getUserID();
         User currentUser = datastoreManager.retrieveUser(userId);
 
-        if (currentUser ==  null) {
-            User user = new User.Builder(userId).build();
-            datastoreManager.createOrReplaceUser(user);
+        if (currentUser !=  null) {
+            response.sendRedirect("/profilePage.html");
+        }
+        
+        String randomUserName = randomUserNameGenerator();
+        while (datastoreManager.doesUsernameExist(randomUserName)) {
+            randomUserName = randomUserNameGenerator();
         }
 
-        response.sendRedirect("/profilePage.html");
+        User user = new User.Builder(userId).setUsername(randomUserName).build();
+        datastoreManager.createOrReplaceUser(user);   
     }
+
+    private String randomUserNameGenerator() {
```

Comment:
Maybe rename to generateRandomUsername(). The name randomUserNameGenerator seems like a value rather than a function.

Reply:
Makes sense, I will change this!